lib/ClangImporter/ImportType.cpp

```diff
@@ -60,6 +60,36 @@ bool ClangImporter::Implementation::isOverAligned(clang::QualType type) {
   return align > clang::CharUnits::fromQuantity(MaximumAlignment);
 }
 
+/// Returns a specialization of the given pointer type for the given pointee.
+///
+/// \p kind must not be a raw pointer kind.
+static Type getPointerTo(Type pointee, PointerTypeKind kind) {
+  ASTContext &ctx = pointee->getASTContext();
+  NominalTypeDecl *pointerDecl = ([&ctx, kind] {
+    switch (kind) {
+    case PTK_UnsafeMutableRawPointer:
+    case PTK_UnsafeRawPointer:
+      llvm_unreachable("these pointer types don't take arguments");
+    case PTK_UnsafePointer:
+      return ctx.getUnsafePointerDecl();
+    case PTK_UnsafeMutablePointer:
+      return ctx.getUnsafeMutablePointerDecl();
+    case PTK_AutoreleasingUnsafeMutablePointer:
+      return ctx.getAutoreleasingUnsafeMutablePointerDecl();
+    }
+    llvm_unreachable("bad kind");
+  }());
+
+  // Specifically handle AUMP being missing to allow testing more of ObjC
+  // interop on non-Apple platforms.
+  assert((pointerDecl || kind == PTK_AutoreleasingUnsafeMutablePointer) &&
+         "could not find standard pointer type");
+  if (!pointerDecl)
+    return Type();
+
+  return BoundGenericType::get(pointerDecl, /*parent*/nullptr, pointee);
```

Comment:
It looks like lib/Sema/TypeCheckType.cpp has a `static Type getPointerType()` helper that looks a lot like this one. Think it's worth extracting a method to a common location? Maybe `Type Type::getPointerType(PointerTypeKind kind)`?

Reply:
Yeah, all right. That is where I got this implementation from, but they were slightly different and…yeah, your way makes sense.

Reply:
Oh right, the TypeChecker one diagnoses if the pointer types can't be found. I don't really think that's so important, though, do you? A bunch of things break if you have a busted stdlib.

Reply:
A little weird to special case AUMP there, though. I'd probably just drop the assert entirely.

Reply:
I think you can justify special-casing AUMP in other places because it genuinely isn’t expected to always exist while the others all should unless the standard library is broken, but dropping the assert would be fine too.